packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -393,6 +393,8 @@ class AppDomain extends Domain {
 
     if (options.debuggingEnabled) {
       connectionInfoCompleter = new Completer<DebugConnectionInfo>();
+      // Callback hookups' Futures are uninteresting.
+      // ignore: unawaited_futures
       connectionInfoCompleter.future.then<Null>((DebugConnectionInfo info) {
```

Comment:
From looking at the code, `connectionInfoCompleter` never completes with an error.

From memory, it's only used to pass along connection info (the service protocol url) when we get it. If the apps fails while starting up, that failure is passed along through other means.